src/features/courses/course-repository.ts

```diff
@@ -544,6 +544,21 @@ class CourseRepository {
         }
     }
 
+    async getStudentTopicOverride(options: {userId: number; topicId: number}): Promise<StudentTopicOverride | null> {
+        try {
+            const override = await StudentTopicOverride.findOne({
+                where: {
+                    userId: options.userId,
+                    courseTopicContentId: options.topicId,
+                    active: true
+                }
+            });
+            return override; // null is fine, there might not *be* any override for this combo...
```

Comment:
I think one cool way to do this would to have a `throw` option in the settings that defaults to true
This way we can default it and if we know it can be null we can set it to false

The problem is if it can be null we will have to add an additional null check, might make sense to just always put that one layer up

Reply:
I think we reconfig our get<object> calls to automatically handle overrides where appropriate, or incorporate a withOverride: boolean

Reply:
Yeah that's on option, my only problem with doing at a super high level is that we might want to do something with the original data

Also once we apply the overrides we modify the model so it becomes in possible to pass it around and use the mixins